spec/unused_i18n_key_checker.rb

```diff
@@ -0,0 +1,35 @@
+class UnusedI18nKeyChecker
+  def self.log(key)
+    @used_keys ||= []
+    @used_keys << key.to_s unless @used_keys.include?(key.to_s)
+  end
+
+  def self.check_unused_keys(ignore_stems: [])
+    return unless ENV["CHECK_UNUSED_KEYS"]
+
+    mappings = YAML.load_file(Rails.root.join("config/locales/en.yml"))
+    defined_keys = define_keys(mappings["en"])
+    not_ignored_defined_keys = defined_keys.reject do |defined_key|
+      ignore_stems.any? { defined_key.starts_with?(_1) }
```

Comment:
A typical i18n stemmer like Snowball will convert "liking" to the stem "like". So `starts_with` wouldn't include it. That may be helpful or maybe I've simply misunderstood what's going on here....!

Reply:
Ah, different sort of "stem" here. This isn't to do with the text content but rather the key. E.g. a typical key might be "activemodel.errors.models.level_of_help_form.attributes.level_of_help.blank", and this line means that if we've passed in `ignore_stems: ["activemodel.error"]` as a parameter, that key would be ignored because it starts with an ignorable stem.